controllers/progressiverollout_controller.go

```diff
@@ -180,3 +197,26 @@ func (r *ProgressiveRolloutReconciler) requestsForSecretChange(o client.Object)
 
 	return requests
 }
+
+// GetTargetClusters returns a list of ArgoCD clusters matching the provided label selector
+func (r *ProgressiveRolloutReconciler) GetTargetClusters(selector metav1.LabelSelector) (corev1.SecretList, error) {
```

Comment:
what is the logic behind returning a list of secrets here instead of a list of clusters?
argo-cd has a specific object `Cluster`, wouldn't that make more sense to use that? It would make the validation with the argocd project easier as well down the line, and manipulating the same types across the ecosystem is a bit more future proof too.

https://github.com/argoproj/argo-cd/blob/12ec99ad01a945e045225e5c239f8ee203e8139a/util/db/cluster.go#L80-L100
https://github.com/argoproj/argo-cd/blob/12ec99ad01a945e045225e5c239f8ee203e8139a/pkg/apis/application/v1alpha1/types.go#L1104-L1126

Reply:
they don't seem to have a selector though :-(

Reply:
Because from the Kubernetes point of view they *are* secrets, the `cluster` object doesn't exist.

Reply:
It is also the same strategy used by the cluster generator in the ApplicationSet, as per https://github.com/argoproj-labs/applicationset/blob/master/pkg/generators/cluster.go

Reply:
OK fine, thanks for pointing me the example on appsets.
we can still unmarshall the secrets and create/reuse a specific cluster type later.